### PR TITLE
Fix 1182 mixed tech ds do not save ba tech type

### PR DESCRIPTION
**Pull request by Sleet01**
Fairly extensive overhaul of Transporter bay encoding / decoding to fix [MML 1182](https://github.com/MegaMek/megameklab/issues/1182) so that mixed-tech vehicles with BA transport bays can utilize the correct type and save/load correctly.

Overview:
1. Replaced the current variable-length "Numbers" field encoding transporter bays with fixed-length encoding that uses 6 fields.  This simplifies Transporter bay loading significantly (when reading the new format).
2. Moved infantry platoon type and facing to separate fields; previously, either could be found in field 2 _or_ 3.
3. Removed 'f' prefix from the "Facing" data, but still allow each entity type to store arbitrary integers in field 4 as there does not appear to be a set of unified "Facing"  values used by all Entities.
4. Made the bay type field (field 5) a bitmap to store various states, e.g. ComStar, Clan, or IS tech base, compactly.
5. Added conversion function to read in older BLK files but store new files in the updated format.
6. Updated all Bay types to output their data in the new format.
7. Added unit tests of the new conversion functions.
8. Added a unit test that verifies that we now save/load data as expected per MML 1182 above. 

New bay "numbers" format:
```
// Field 0 is the size of the bay, in tons or # of units and is required
// Field 1 is the number of doors in the bay, and is required
// Field 2 is the bay number and is required; default value of "-1" is "unset"
// Field 3 is used to record infantry platoon type; default is an empty string
// Field 4 is an int recording facing; default is string representation of entity.LOC_NONE
// Field 5 is a bitmap recording status like tech type, ComStar bay; default is "0"
```

Compatibility:
- 49.14 BLK file -> 49.XX load: OK
- 49.XX BLK file -> 49.XX load: OK
- 49.XX BLK file -> 49.14 (or earlier) load: Entities with bays will load but some data may be incorrect, specifically mixed-tech BA bay weights (they will all be converted to the base tech type), ComStar BA bays (won't be correct type), and any bays with facings (e.g. in Space Stations).

### CODE REVIEW

**Comment by Sleet01** [comment]
I'll update the PR to fix the above issues tonight.

File: megamek/src/megamek/common/loaders/BLKFile.java [hunk 1]
```diff
@@ -1243,94 +1275,184 @@
...
+                    bitmap |= COMSTAR_BIT;
+                }
+                else if (
+                        Set.of(
+                            new String [] {"jump", "foot", "motorized", "mechanized"}
+                        ).contains(potentialBayTypeIndicator.toLowerCase())){
+                    // Found an infantry type in the last field (2 or 3)
+                    // Assign to field 3
+                    temp[3] = potentialBayTypeIndicator;
+                    if (temp[2] == temp[3]){
```
[the remark this answers is not in the document]
Comment (Sleet01) [reply megamek/src/megamek/common/loaders/BLKFile.java:1424]: This works in Python ^_^

Will fix.

File: megamek/unittests/megamek/common/loaders/BLKDropshipFileTest.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        boolean mixedTech = false;
+        boolean ISBACorrect = false;
+        boolean ClanBACorrect = false;
+        boolean ComStarBACorrect = false;
+        Vector<Bay> bays = null;
+
+        try{
+            Dropship ds = loadDropshipFromString(newFormatDSwithMixedBA);
+            parsed = true;
+            mixedTech = ds.isMixedTech() & ds.isClan(); // confirm mixed-tech Clan design
```
[the remark this answers is not in the document]
Comment (Sleet01) [reply megamek/unittests/megamek/common/loaders/BLKDropshipFileTest.java:210]: I see it; will fix.

**Review by NickAragua — COMMENTED** [review]
Does this do ok loading both "before" and "after" blk files?
Edit: reading the PR notes, apparently mostly yes.

**Review by NickAragua — APPROVED** [review]
Looks fine, but I'll leave this up for a little bit for others to take a look at as it is fairly extensive.

**Comment by HammerGS** [comment]
Merging so QA has access to test.

**Pull request merged**